### ISSUE TITLE
sensitive mode produces fewer HSPs than default mode

### ISSUE THREAD

**Issue by xinehc** [bbuchfink/diamond#622]
Hi,

I was trying to map some very long contigs (mostly complete genomes) to a customized protein database using diamond blastx (with `--range-culling`).

Using the `default` mode I got 29453 mapped queries and 1092752 HSPs.
```
diamond blastx --db $s.dmnd --query $q.fa --outfmt 6 --evalue 10 --range-culling -k1 -F 15 --threads 64 --out ${q}-${s}-default.txt

Total time = 5111.44s
Reported 1092752 pairwise alignments, 1092752 HSPs.
29453 queries aligned.
```

After switching to the `sensitive` mode the number of mapped queries increased but HSPs decreased.

```
diamond blastx --db $s.dmnd --query $q.fa --outfmt 6 --evalue 10 --range-culling -k1 -F 15 --threads 64 --sensitive --out ${q}-${s}-sensitve.txt

Total time = 14308.2s
Reported 1078610 pairwise alignments, 1078610 HSPs.
43929 queries aligned.
```
I was wondering why the sensitive mode produces fewer HSPs? Is this behavior related to `--range-culling` or `--min-orf`? For those long contigs would diamond be more accurate if I first predict the ORFs and then use blastp instead of directly using blastx with `--range-culling`?

Thanks!

**Comment by xinehc** [comment]
Thank you very much for your explanation!

Here is one example (query and db attached):

```
diamond blastx --db ko40.dmnd --query GCF_001273775.1.fa --outfmt 6 --evalue 10 --range-culling -k1 -F 15 --threads 64 --out default.txt

Total time = 2.351s
Reported 41 pairwise alignments, 41 HSPs.
1 queries aligned.
```

```
diamond blastx --db ko40.dmnd --query GCF_001273775.1.fa --outfmt 6 --evalue 10 --range-culling -k1 -F 15 --threads 64 --out sensitive.txt --sensitive

Total time = 6.919s
Reported 40 pairwise alignments, 40 HSPs.
1 queries aligned.
```
I thought this could be caused by overlapping query ranges so I manually checked the HSPs and found that they were completely separated (0% overlap) in both modes. 

Then I tuned `--range-cover`. When setting it to a very low value (0.0001), although the number of HSPs didn't change, they were not completely identical to the default ones. So as a follow-up question I was wondering why `--range-cover` matters even if all HSPs do not overlap?

```
diamond blastx --db ko40.dmnd --query GCF_001273775.1.fa --outfmt 6 --evalue 10 --range-culling -k1 -F 15 --threads 64 --out default-noRangeCover.txt --range-cover 0.0001

Total time = 2.54s
Reported 41 pairwise alignments, 41 HSPs.
1 queries aligned.
```

Thanks!

[test.zip](https://github.com/bbuchfink/diamond/files/10015288/test.zip)

**Comment by bbuchfink** [comment]
> I noticed from [#575 (comment)](https://github.com/bbuchfink/diamond/issues/575#issue-1205475721) that if a target sequence appears multiple times in the query (which is likely for long query sequences), only a single HSPs will be reported and `--max-hsps` will not function.

This was actually just the hit culling being too agressive, deleting hits if the query or target ranges overlap. This should fix it: 1938c340069f778c55815d9965d328b4a160e7fa

**Comment by xinehc** [comment]
Thanks! This magic patch solved all the problems.
